### DISCUSSION TITLE
How to cache images on Azure storage

### DISCUSSION THREAD

**Discussion by srdjo** [SixLabors/ImageSharp.Web#101] [Q&A] [answered]
Hi,

I am trying to configure ImageSharp.Web with AzureBlobStorageImageProvider and all is working fine but is there a way to cache images on the same Azure storage and not in the local file system ?

I am migrating my ASP.Net MVC app to Asp.net Core 3.1 app with ServiceStack and Nuxt.js front end that is hosted on Azure and I would like to prevent cache files clogging my application.

Also is there a way to make named presets like there was in ImageProcessor.Web ?

Best regards.

**Comment by JimBobSquarePants** [comment, marked as answer]
>I am trying to configure ImageSharp.Web with AzureBlobStorageImageProvider and all is working fine but is there a way to cache images on the same Azure storage and not in the local file system ?

You have to implement and register your own [`IImageCache`](https://docs.sixlabors.com/api/ImageSharp.Web/SixLabors.ImageSharp.Web.Caching.IImageCache.html)

You can use the [`PhysicalFileSystemCache`](https://github.com/SixLabors/ImageSharp.Web/blob/2bce8036b34dd4872911fbff816dba07a9d690e2/src/ImageSharp.Web/Caching/PhysicalFileSystemCache.cs) for inspiration.

>Also is there a way to make named presets like there was in ImageProcessor.Web ?

There's two ways:
1. Implement and register your own [`IImageWebProcessor`](https://docs.sixlabors.com/api/ImageSharp.Web/SixLabors.ImageSharp.Web.Processors.IImageWebProcessor.html) that captures a single query and converts it into a predefined process.
2. Use [ASP.NET Core TagHelpers](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/tag-helpers/intro?view=aspnetcore-3.1) to render the correct querystring command based upon your own application logic.

**Reply by JimBobSquarePants** [reply]
@srdjo An official blob storage cache is coming with V1. Stay tuned!

**Comment by srdjo** [comment]
Can you give me more information on how to implement and register processors and providers ?

I just need to enable Entropycrop since I used it in ImageProcessor.Web

Thanks in advance

**Reply by JimBobSquarePants** [reply]
The best way to do it is to copy  how existing processors do it, you can find the source code  [here](https://github.com/SixLabors/ImageSharp.Web/tree/2bce8036b34dd4872911fbff816dba07a9d690e2/src/ImageSharp.Web/Processors). Registration is documented in the readme and [docs](https://docs.sixlabors.com/articles/imagesharp.web/gettingstarted.html).
